### PR TITLE
Hint text should wrap with multiple lines TextField

### LINKED ISSUE #21045: Hint text in TextField should wrap when the field is longer than 1 line
This was brought up [here](https://github.com/flutter/flutter/issues/11197) but there has been no activity on that issue for over a year.

### PR DESCRIPTION
**Pull request by tran-huy-phuc**
Fix #21045 #11197

### CODE REVIEW

File: packages/flutter/lib/src/material/text_field.dart [hunk 1]
```diff
@@ -582,6 +582,7 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
             isFocused: focusNode.hasFocus,
             isEmpty: controller.value.text.isEmpty,
             child: child,
+            hintTextOverflow: (widget.maxLines == null || widget.maxLines <= 1) ? TextOverflow.ellipsis : TextOverflow.clip,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/text_field.dart]: I don't think we should specify `hintTextOverflow` in this case; clipping overflow text can be difficult for users to interpret.

**Review by HansMuller — COMMENTED** [review]
I'm not convinced that this is a net-positive change. We should probably add a widget-valued hint property to InputDecorator instead of continuing to add API for configuring the hint's Text widget (or the error's text widget, etc). As far as overflow-handling goes, I think what's really needed is support for adding an ellipsis when the text overflows maxLines.

There are some difficult outstanding problems around "eillipsizing" text. See for example:
- https://github.com/flutter/flutter/issues/15465
- https://github.com/flutter/flutter/issues/18761

**Comment by thphuc** [comment]
Thanks @HansMuller for your review and feedback. Should I continue with this approach or find and switch to another solution?

**Comment by HansMuller** [comment]
@thphuc - I'm afraid the generalized multiline ellipsis solution may be a ways off. Probably the best way to address this now would be to add a widget valued `hint` parameter to InputDecorator. Doing so will require adding tests which verify that the common uses cases like yours layout, animate, and paint correctly.

**Comment by Hixie** [comment]
@thphuc Are you interested in updating this PR to use the approach Hans suggested, or would you rather we close this PR and work on the solution later? Either way is fine, let me know! Thanks for your contributions so far!

**Comment by thphuc** [comment]
@Hixie Thank you for your reminder. I would like to update this PR by adding a "hint" widget to InputDecorator. Before doing that, do you have any advice/suggestion (assert, write unit test...)?

**Comment by thphuc** [comment]
Thanks @HansMuller @Hixie I will need more time to understand the structure of Flutter framework. I will come back to this issue when I feel more confident. Thanks!

**Comment by dnfield** [comment]
@thphuc Thanks for the efforts here!

I'm going to close this for now to get it off of review queues, but please feel free to reopen if you intend to work on it further to address some of the issues raised here.

**Pull request closed without merging**